nomad/resource_job_v2.go

```diff
@@ -0,0 +1,1682 @@
+	}
+
+	return nil, nil
+}
+
+func getUpdate(d interface{}) (*api.UpdateStrategy, error) {
+	for _, up := range d.([]interface{}) {
+		u := up.(map[string]interface{})
+
+		update := &api.UpdateStrategy{
+			MaxParallel: getInt(u, "max_parallel"),
+			HealthCheck: getString(u, "health_check"),
+			Canary:      getInt(u, "canary"),
+			AutoRevert:  getBool(u, "auto_revert"),
+			AutoPromote: getBool(u, "auto_promote"),
+		}
+
+		var duration *time.Duration
+		duration, err := getDuration(u["stagger"])
+		if err != nil {
+			return nil, err
+		}
+		update.Stagger = duration
+
+		duration, err = getDuration(u["min_healthy_time"])
+		if err != nil {
+			return nil, err
+		}
+		update.MinHealthyTime = duration
+
+		duration, err = getDuration(u["healthy_deadline"])
+		if err != nil {
+			return nil, err
+		}
+		update.HealthyDeadline = duration
+
+		duration, err = getDuration(u["progress_deadline"])
+		if err != nil {
+			return nil, err
+		}
+		update.ProgressDeadline = duration
+
+		return update, nil
+	}
+	return nil, nil
+}
+
+func getTasks(d interface{}) ([]*api.Task, error) {
+	tasks := make([]*api.Task, 0)
+
+	for _, tk := range d.([]interface{}) {
+		t := tk.(map[string]interface{})
+
+		artifacts := make([]*api.TaskArtifact, 0)
+		for _, af := range t["artifact"].([]interface{}) {
+			a := af.(map[string]interface{})
+
+			artifact := &api.TaskArtifact{
+				GetterSource:  getString(a, "source"),
+				GetterOptions: getMapOfString(a["options"]),
+				GetterMode:    getString(a, "mode"),
+				RelativeDest:  getString(a, "destination"),
+			}
+
+			artifacts = append(artifacts, artifact)
+		}
+
+		var dispatchPayloadConfig *api.DispatchPayloadConfig
+		for _, dp := range t["dispatch_payload"].([]interface{}) {
+			d := dp.(map[string]interface{})
+
+			dispatchPayloadConfig = &api.DispatchPayloadConfig{
+				File: d["file"].(string),
+			}
+		}
+
+		var taskLifecycle *api.TaskLifecycle
+		for _, tl := range t["lifecycle"].([]interface{}) {
+			l := tl.(map[string]interface{})
+
+			taskLifecycle = &api.TaskLifecycle{
+				Hook:    l["hook"].(string),
+				Sidecar: l["sidecar"].(bool),
+			}
+		}
+
+		templates := make([]*api.Template, 0)
+		for _, tpl := range t["template"].([]interface{}) {
+			tp := tpl.(map[string]interface{})
+
+			template := &api.Template{
+				ChangeMode:   getString(tp, "change_mode"),
+				ChangeSignal: getString(tp, "change_signal"),
+				EmbeddedTmpl: getString(tp, "data"),
+				DestPath:     getString(tp, "destination"),
+				Envvars:      getBool(tp, "env"),
+				LeftDelim:    getString(tp, "left_delimiter"),
+				Perms:        getString(tp, "perms"),
+				RightDelim:   getString(tp, "right_delimiter"),
+				SourcePath:   getString(tp, "source"),
+			}
+
+			duration, err := getDuration(tp["splay"])
+			if err != nil {
+				return nil, err
+			}
+			template.Splay = duration
+
+			duration, err = getDuration(tp["vault_grace"])
+			if err != nil {
+				return nil, err
+			}
+			template.VaultGrace = duration
```

Comment:
I think it would be better to avoid re-using the `duration` variable here. It could lead to bugs in the future.